### PR TITLE
Add SONY Rhine and pm8941 SPMI regulators

### PR DESCRIPTION
**Pull request by konradybcio**

### CODE REVIEW

**Comment by konradybcio** [comment]
Sadly no display, nobody cared enough to enable display in the original BL, but hey, at least the eMMC isn't suicidal..

File: dts/msm8974/msm8974-sony-xperia-rhine.dts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+/include/ "skeleton.dtsi"
+
+/ {
+	model = "SONY Xperia Z1 / Z1 Compact / Z Ultra";
+	compatible = "somc,rhine", "qcom,msm8974", "lk2nd,device";
+	// thanks for giving us device-unique IDs, oh wait..
+	lk2nd,match-bootloader = "*Rhine*";
+
+	// You are ID 8, you are ID 8, EVERYBODY IS ID 8!
+	qcom,msm-id = <0 0>; /* Rhine bootloader doesn't care but dtbTool apparently does */
```
Comment (Minecrell) [inline dts/msm8974/msm8974-sony-xperia-rhine.dts:14]: Would prefer if you add some useful msm-id here if bootloader doesn't care.

File: dts/msm8974/msm8974-sony-xperia-rhine.dts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	qcom,board-id = <8 0>;
```
Comment (Minecrell) [inline dts/msm8974/msm8974-sony-xperia-rhine.dts:15]: I guess you will need to coordinate with @z3ntu (#69) and add a generic dts with subnodes for msm8974-mtp I guess.

**Comment by z3ntu** [comment]
This was my solution which also covers pm8841 and pma8084 https://github.com/z3ntu/lk2nd/commit/5e147f7056ee0b080a5e224c68de6927a212288a

**Comment by minlexx** [comment]
WHy not have separate structs for each?

**Comment by z3ntu** [comment]
If you mean pma8084 vs pm8941, you'd need to query the pmic id from smem and choose the correct struct based on that - which is more complicated.

**Comment by minlexx** [comment]
lk already does this (parse smem etc)

**Comment by TravMurav** [comment]
Sorry, this PR got a bit stale...
Lately we were trying to refactor lk2nd so I'm not sure if this change is still relevant thus I'm closing it. If you want to respin it, please feel free to reopen or re-target to the `experimental-tmp2` branch.

**Pull request closed without merging**